common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -3098,6 +3098,10 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "This is set to compactor from within the query based compactor. This enables the Tez SplitGrouper "
         + "to group splits based on their bucket number, so that all rows from different bucket files "
         + " for the same bucket number can end up in the same bucket file after the compaction."),
+    // MetastoreConf doesn't have a restricted list. Otherwise can be moved there.
+    COMPACTOR_RUN_AS_USER("hive.compactor.run.as.user", "",
+        "Specify the user to run compaction (Worker) and compaction-related file listing (Initiator, Cleaner) as. " +
+        "If empty string, defaults to table/partition directory owner. Restricted."),
```

Comment:
This will be used on HMS and occasionally on HS2 (if the Worker is HS2).
Where are those other Compaction related configs? If everything is on HMS side, please move this there too

Reply:
MetastoreConf, but it has no restricted list AFAIK. Implementing one would be a new feature that isn't covered by this ticket.

Reply:
AFAIK only very specific HMS confs could be changed by the user (HiveConf.metaConfVars). So restricting it might not be needed.